Z-Wave Basic Window Shade fix + Calibration

Aeotec-ccheng opened the pull request:
- Open() and Close() buttons resolved
- Calibration Parameter setting included in preference

Aeotec-ccheng committed:
Z-Wave Basic Window Shade fix + Calibration

- Open() and Close() buttons resolved
- Calibration Parameter setting included in preference

KKlimczukS commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -25,10 +25,10 @@ metadata {
-		fingerprint mfr:"0086", prod:"0103", model:"008D", deviceJoinName: "Aeotec Window Treatment" //Aeotec Nano Shutter
-		fingerprint mfr:"0371", prod:"0003", model:"008D", deviceJoinName: "Aeotec Window Treatment" //Aeotec Nano Shutter
-		fingerprint mfr:"0371", prod:"0103", model:"008D", deviceJoinName: "Aeotec Window Treatment" //Aeotec Nano Shutter
+		fingerprint mfr:"0086", prod:"0003", model:"008D", deviceJoinName: "Aeotec Nano Shutter" //Aeotec Nano Shutter
```
According to the new policy deviceJoinName should stay as it is now -> Aeotec Window Treatment

Aeotec-ccheng replied:
@KKlimczukS - changed name back to Aeotec Window Treatment.

KKlimczukS commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -58,11 +58,25 @@ metadata {
 				displayDuringSetup: false
 			)
+            
+                input(title: "Aeotec Nano Shutter settings",
```
Please, fix spacing: 4 spaces -> 1 tab (applies to the whole file)

Aeotec-ccheng replied:
@KKlimczukS - Fixed spacing

KKlimczukS commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -147,6 +162,14 @@ def installed() {
+        response([
+    		secure(zwave.configurationV1.configurationSet(parameterNumber: 35, size: 1, scaledConfigurationValue: time)),
+        ])
+    } else {}
```
Please remove the empty "else" block.

Aeotec-ccheng replied:
@KKlimczukS - removed else block on line 171

Aeotec-ccheng committed:
Fix of requested issues

- Return pair name to "Aeotec Window Treatment"
- Fixed spacing of lines 61 - 73
- Removed else block on line 171

dkirker commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -101,11 +115,12 @@ def zwaveEvent(physicalgraph.zwave.Command cmd) {
 			open()
 			break
-		case "close":
+		case "statelessCurtainPowerButton_close_button":
```
These aren't the correct values for this capability. They should be "open" and "close". The capability's command definition is as follows:
```
commands:
  setButton:
    arguments:
    - name: button
      required: true
      schema:
        type: string
        enum:
          - open
          - close
          - pause
      type: ENUM
      values:
        - open
        - close
        - pause
```

If this value is difference then we should investigate why they are.

Aeotec-ccheng replied:
@dkirker - When i started working with Nano Shutter using the default device handler, it wasn't working initially.

I soon found that all values were parsing to Pause() only.

I used log.debug "button: $button" (shown on line 118) to find the states;

"statelessCurtainPowerButton_open_button" and "statelessCurtainPowerButton_close_button" which were reporting when pressing open or close from SmartThings connect.

If you run my code, you should be able to use line 118 log.debug to see the specific state that is reported.

dkirker replied:
@Aeotec-ccheng That's what I was suspecting had happened. Thanks for confirming. Let's go ahead and do the following so that when the values that should be sent are sent the code will still work:
```
		case "open":
		case "statelessCurtainPowerButton_open_button":
			open()
			break
		case "close":
		case "statelessCurtainPowerButton_close_button":
			close()
			break
```

Aeotec-ccheng replied:
@dkirker - Thanks for the coding recommendation, added that in to the new commit.

dkirker replied:
@Aeotec-ccheng Awesomesauce!

dkirker commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -147,6 +162,14 @@ def installed() {
 def updated() {
 	sendHubCommand(pause())
 	state.reverseDirection = reverseDirection ? reverseDirection : false
+    int time = calibrationTime
```
Please use tabs instead of spaces for indentation. (There are some spaces in the preference definition block above, as well.)

Aeotec-ccheng replied:
@dkirker - fixed

greens commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -58,11 +58,25 @@ metadata {
+			    element: "paragraph")
+
+			input("calibrationTime", "number",
+			    title: "Set the open/close time for motor, range = 5 - 255",
```
you could add a `range: "5..255"` field instead of this text

Aeotec-ccheng replied:
@greens - fixed

Aeotec-ccheng committed:
Fixed Requested issues

- Fixed indentation to use tabs. (line 165)
- Fixed range instead of the use of text. (line 69)

greens commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -147,13 +163,21 @@ def installed() {
 	state.reverseDirection = reverseDirection ? reverseDirection : false
+	int time = calibrationTime
+
+	if (calibrationTime >= 5 && calibrationTime <= 255) {
```
did you mean to be checking `time` instead of `calibrationTime` here? As far as I can tell they should be equal, though, so I don't know if having a second variable is worthwhile.

cbaumler replied:
I agree that we should just use `calibrationTime` here.

Aeotec-ccheng replied:
@greens @cbaumler - the "int time = calibrationTime" was primarily with me messing around and practicing a little bit while figuring out how preferences are used as numbers/integrrs, i ended up leaving it in there and forgot to remove it when i realized calibrationTime was pretty much a local variable for the device handler.

I'll have that removed.

Aeotec-ccheng replied:
@greens @cbaumler - edited for this function to only use calibrationTime number.

greens commented on devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:
```diff
@@ -58,11 +58,26 @@ metadata {
+			    defaultValue: false,
+			    displayDuringSetup: false,
+			    range: "5..255",
+			    default: 10
```
While you're here, could you just refactor how all of these settings are presented? I think we could consolidate them quite a bit.

I'm thinking it should just be:
```
			input("reverseDirection", "bool",
				title: "Reverse working direction",
                                 description: "In case wiring is wrong, this setting can be changed to fix setup without any manual maintenance.",
				defaultValue: false,
				displayDuringSetup: false
			)            	

			input("calibrationTime", "number",
                             title: "Open/Close timing"
			    description: "Set the motor's open/close time",
			    defaultValue: false,
			    displayDuringSetup: false,
			    range: "5..255",
			    default: 10
			)
```
(but don't just copy/paste that because the spacing will be off)

I'd also add a small comment next to both of these just noting that they're for the Aeotec Nano Shutters in case we ever add another device to this DTH.

Aeotec-ccheng replied:
@Greens - not a problem here will have this edited soon.

Aeotec-ccheng replied:
@greens - Edited and setting this to resolved.

Aeotec-ccheng committed:
Update Part 3 - Nano Shutter

- used calibrationTime rather than int Time. (Line 164)
- Condensed preference input for "calibrationTime"
- Added note on top of calibrationTime to indicate it is specific to Nano Shutter and the function its used in.

Aeotec-ccheng committed:
Update Nano Shutter Part 4

- Added
case "open": on line 117
case "close": on line 121

greens reviewed (approved):
lgtm

dkirker reviewed (approved):
Looks good!
